lib/git/base.rb

```diff
@@ -35,56 +27,82 @@ def self.config
       return @@config ||= Config.new
     end
 
-    # initializes a git repository
-    #
-    # options:
-    #  :bare
-    #  :index
-    #  :repository
-    #
-    def self.init(working_dir, opts = {})
-      opts[:working_directory] ||= working_dir
-      opts[:repository] ||= File.join(opts[:working_directory], '.git')
+    # (see Git.init)
+    def self.init(directory, options = {})
+      options[:working_directory] ||= directory
+      options[:repository] ||= File.join(options[:working_directory], '.git')
 
-      FileUtils.mkdir_p(opts[:working_directory]) if opts[:working_directory] && !File.directory?(opts[:working_directory])
+      FileUtils.mkdir_p(options[:working_directory]) if options[:working_directory] && !File.directory?(options[:working_directory])
 
-      init_opts = {
-        :bare => opts[:bare]
-      }
+      init_options = { :bare => options[:bare] }
 
-      opts.delete(:working_directory) if opts[:bare]
+      options.delete(:working_directory) if options[:bare]
 
       # Submodules have a .git *file* not a .git folder.
       # This file's contents point to the location of
       # where the git refs are held (In the parent repo)
-      if opts[:working_directory] && File.file?(File.join(opts[:working_directory], '.git'))
+      if options[:working_directory] && File.file?(File.join(options[:working_directory], '.git'))
         git_file = File.open('.git').read[8..-1].strip
-        opts[:repository] = git_file
-        opts[:index] = git_file + '/index'
+        options[:repository] = git_file
+        options[:index] = git_file + '/index'
       end
 
-      Git::Lib.new(opts).init(init_opts)
+      # TODO: this dance seems awkward: this creates a Git::Lib so we can call
+      #   init so we can create a new Git::Base which in turn (ultimately)
+      #   creates another/different Git::Lib.
+      #
+      # TODO: maybe refactor so this Git::Bare.init does this:
+      #   self.new(opts).init(init_opts) and move all/some of this code into
+      #   Git::Bare#init. This way the init method can be called on any
+      #   repository you have a Git::Base instance for.  This would not
+      #   change the existing interface (other than adding to it).
+      #
+      Git::Lib.new(options).init(init_options)
 
-      self.new(opts)
+      self.new(options)
     end
 
-    # opens a new Git Project from a working directory
-    # you can specify non-standard git_dir and index file in the options
-    def self.open(working_dir, opts={})
-      opts[:working_directory] ||= working_dir
-      opts[:repository] ||= File.join(opts[:working_directory], '.git')
+    # (see Git.open)
+    def self.open(working_dir, options={})
```

Comment:
The arg opts was changed to options to be the same as `Git.open` so the `(see Git.open)` YARD directive would work.